frontend/src/components/Header.js

```diff
@@ -15,62 +17,91 @@ const Header = () => {
   const logoutHandler = () => {
     dispatch(logout());
   };
-
   return (
-    <header>
-      <Navbar bg="dark" variant="dark" expand="lg" collapseOnSelect>
-        <Container>
-          <LinkContainer to="/">
-            <Navbar.Brand>ProShop</Navbar.Brand>
-          </LinkContainer>
-          <Navbar.Toggle aria-controls="basic-navbar-nav" />
-          <Navbar.Collapse id="basic-navbar-nav">
-            <Route
-              render={({ history }) => (
-                <SearchBox history={history} />
-              )}
-            />
-            <Nav className="ml-auto">
-              <LinkContainer to="/cart">
-                <Nav.Link>
-                  <i className="fas fa-shopping-cart" /> Cart
-                </Nav.Link>
-              </LinkContainer>
-              {userInfo ? (
-                <NavDropdown title={userInfo.name} id="username">
-                  <LinkContainer to="/profile">
-                    <NavDropdown.Item>Profile</NavDropdown.Item>
-                  </LinkContainer>
-                  <NavDropdown.Item onClick={logoutHandler}>
-                    Logout
-                  </NavDropdown.Item>
-                </NavDropdown>
-              ) : (
-                <LinkContainer to="/login">
-                  <Nav.Link>
-                    <i className="fas fa-user" /> Sign In
-                  </Nav.Link>
-                </LinkContainer>
-              )}
-              {userInfo && userInfo.isAdmin && (
-                <NavDropdown title="Admin" id="adminmenu">
-                  <LinkContainer to="/admin/userlist">
-                    <NavDropdown.Item>Users</NavDropdown.Item>
-                  </LinkContainer>
-                  <LinkContainer to="/admin/productlist">
-                    <NavDropdown.Item>Products</NavDropdown.Item>
-                  </LinkContainer>
-                  <LinkContainer to="/admin/orderlist">
-                    <NavDropdown.Item>Orders</NavDropdown.Item>
-                  </LinkContainer>
-                </NavDropdown>
-              )}
-            </Nav>
-          </Navbar.Collapse>
-        </Container>
-      </Navbar>
-    </header>
+    <>
+      <header>
+        <StyledNavbar expand="lg" collapseOnSelect variant="dark">
+          <Container fluid>
+            <LinkContainer to="/">
+              <Navbar.Brand>
+                <StyledH1>ProShop</StyledH1>
+              </Navbar.Brand>
+            </LinkContainer>
+            <Navbar.Toggle aria-controls="basic-navbar-nav" />
+            <Navbar.Collapse id="basic-navbar-nav">
+              <Route
+                render={({ history }) => (
+                  <SearchBox history={history} />
+                )}
+              />
+              <Nav className="ml-auto">
+                <StyledA
+                  href={userInfo ? '/cart' : '/register'}
+                  className="navLinks"
+                >
```

Comment:
Use the styled nav link component here. Using this directly causes the page to reload, hence use the "as" prop, and assign it the Link component of react router. And use "to" instead of "href"
Eg:-  <Nav.Link as={Link} to="/" >